src/app/api/attachments/attachments.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { CreateAttachmentRequest, CreateAttachmentRequestSchema } from '@/types/dto/attachments.dto'
+import { AttachmentsService } from '@api/attachments/attachments.service'
+import httpStatus from 'http-status'
+import { handleUpload, type HandleUploadBody } from '@vercel/blob/client'
+import { IdParams } from '@api/core/types/api'
+
+export const createAttachment = async (request: NextRequest) => {
+  const user = await authenticate(request)
+  const body = (await request.json()) as HandleUploadBody
+  const attachmentsService = new AttachmentsService(user)
+  const fileUploadResponse = await handleUpload({
+    body,
+    request,
+    onBeforeGenerateToken: async (pathname: string, clientPayload: string | null) => {
+      if (!user.internalUserId) {
+        throw new Error('Not authorized')
+      }
```

Comment:
Throw `APIError` instead. Else we will return a `500` server side error http status even if this is supposed to be a `401`